detectors/node/opentelemetry-resource-detector-gcp/test/detectors/GcpDetector.test.ts

```diff
@@ -167,8 +179,77 @@ describe('gcpDetector', () => {
     });
 
     it('returns empty resource if not detected', async () => {
-      const resource = await gcpDetector.detect();
+      const resource = gcpDetector.detect();
+      await resource.waitForAsyncAttributes?.();
+
       assertEmptyResource(resource);
     });
   });
+
+  describe('internal tracing', () => {
```

Comment:
(I guess I forgot to hit "save" in my first review.)

Wow this was a fair amount of effort to write a test case for this.
I don't know how others would feel, but I wouldn't require similar test code be added for suppressTracing usage in other detectors.

Reply:
Yup it was a fair amount of work 😓. But I needed
- the `sdk` running so I had to create the fixture
- to avoid `nock` since real requests won't happen so we won't be really testing
- `gcp-metadata` to get responses. I could let it fail but I found that the request time vary in a way that makes the test flaky.

there is another approach I can take but 1st I wanted some feeedback for this one. If we finally use this one I may be inclined to move the mock server to `@opentelemetry/contrib-test-utils`

Reply:
About the 3rd point in my previous comment I found that if the async attributes of the resource are resolved after calling `sdk.shutdown()` we loose some spans. Without knowing to much about the mechanics of the processors/exporters I'd guess that these spans are not yet in the queue that is flushed during shutdown. I think this could be a potential issue for apps that have a short execution time (thinking on lamba or similar)

Reply:
@trentm the other approach seems to work and it's shorter 🎉 (although we're getting a resource with empty values). Changes are
- remove the mock server
- trigger the detection again and `waitForAsyncAttributes` 

```diff
diff --git a/detectors/node/opentelemetry-resource-detector-gcp/test/fixtures/use-gcp-detector.js b/detectors/node/opentelemetry-resource-detector-gcp/test/fixtures/use-gcp-detector.js
index 0c71a081..f88d2ae4 100644
--- a/detectors/node/opentelemetry-resource-detector-gcp/test/fixtures/use-gcp-detector.js
+++ b/detectors/node/opentelemetry-resource-detector-gcp/test/fixtures/use-gcp-detector.js
@@ -78,6 +78,9 @@ server.listen(0, '127.0.0.1', async function () {
   });

   // flush any left spans
+  // NOTE: this adds extra requests but its necessary to make sure
+  // spans have the resouce and are queued in the exporter
+  await gcpDetector.detect().waitForAsyncAttributes();
   await sdk.shutdown();
   await new Promise(resolve => server.close(resolve));
 });
```

A couple of observations:
- test becomes slower since the requests are made again at the end of the fixture code
- this detector just falls back to empty string if the request fails but I'm not sure if this will work on other detectors